cli/tests/integrations/test_config.py

```diff
@@ -66,12 +66,15 @@ def test_get_missing_property(env):
     _get_missing_value('missing.property', env)
 
 
-def test_invalid_dcos_url(env):
-    stderr = b'Please check url \'abc.com\'. Missing http(s)://\n'
-    assert_command(['dcos', 'config', 'set', 'core.dcos_url', 'abc.com'],
-                   stderr=stderr,
-                   returncode=1,
+def test_dcos_url_without_scheme(env):
+    old = b'http://dcos.snakeoil.mesosphere.com'
+    new = b"abc.com"
+    out = b"[core.dcos_url]: changed from '%s' to 'https://%s'\n" % (old, new)
```

Comment:
I asked because I don't like to seeing byte literals in Python code w/o _obvious_ reason :-) We can leave this as is, but I want to elaborate on my point, because I think it's a pretty damn important point in Python application design.

The thumb rule is to use _text_ processing _within_ the application, and to decode and encode _only_ at its boundaries where the IO is done (the communication with the subprocess in this case). That usually makes code way easier to understand, because the application code then usually only contains text objects (`str` in Python 3). If one does not follow that thumb rule, one ends up having a (randomly appearing) mixture of bytes/str objects within the application code, and it's hard to follow the _assumptions_ associated with the corresponding encoding and decoding approaches. The last sentence is really important. To clarify, I want to point out how exactly this code here relies out fancy stuff (which is non-obvious from the code):
* We rely on  `b'...'` literals to magically use the ASCII codec (try `b'A' == b'\x41'`).
* We rely on the subprocess `stdout` (type `bytes`) to be _compatible_ with the ASCII codec (we do a bytes comparison of encoded text, so we require the encoding to be the same in both operands of the comparison). Which _seems_ to be the case here, but I am quite sure it's not true in all cases: _most likely_ the subprocess uses UTF-8 to encode it's text when it emits it to stdout. And ASCII is a subset of UTF-8. That's why the comparison _seems_ to work here.

To make this clean, explicit, and expressive, a good strategy is to _explicitly instruct_ the invoked subprocess (which is the CPython interpreter here) how to encode its data on stdout (if that's Python, this can be done using the environment variable `PYTHONIOENCODING`). Imagine we set this to `UTF-8`. Then we have exact knowledge about how to interpret the data we receive from that subprocess (and can do something like `subprocess_stdout = subprocess_stdout_bytes.decode('utf-8')`). This we could then compare to the `str`(!) object containing `[core.dcos_url]: changed from '%s' to 'https://%s'\n" % (old, new)`.